godot-core/src/builtin/meta/signature.rs

```diff
@@ -74,40 +76,31 @@ macro_rules! impl_varcall_signature_for_tuple {
             where $R: VariantMetadata + ToVariant + sys::GodotFuncMarshal + Debug,
                $( $Pn: VariantMetadata + FromVariant + sys::GodotFuncMarshal + Debug, )*
         {
-            #[inline]
-            fn variant_type(index: i32) -> sys::VariantType {
+            const ARGUMENTS_LEN: usize = $ARGUMENTS_LEN;
+
+            fn argument_info(index: usize, param_name: &str) -> Option<MethodArgumentInfo> {
                 match index {
-                    -1 => $R::variant_type(),
                     $(
-                        $n => $Pn::variant_type(),
+                        $n => Some($Pn::argument_info(param_name)),
                     )*
-                    _ => unreachable!("variant_type: unavailable for index {}", index),
-                    //_ => sys::GDEXTENSION_VARIANT_TYPE_NIL
+                    _ => None,
                 }
             }
 
-            #[inline]
-            fn param_metadata(index: i32) -> sys::GDExtensionClassMethodArgumentMetadata {
-                match index {
-                    -1 => $R::param_metadata(),
-                    $(
-                        $n => $Pn::param_metadata(),
-                    )*
-                    _ => unreachable!("param_metadata: unavailable for index {}", index),
-                }
+            fn return_info() -> Option<MethodArgumentInfo> {
+                $R::return_info()
             }
 
-            #[inline]
-            fn property_info(index: i32, param_name: &str) -> PropertyInfo {
+            fn argument_property_info(index: usize, param_name: &str) -> PropertyInfo {
                 match index {
-                    -1 => $R::property_info(param_name),
                     $(
                         $n => $Pn::property_info(param_name),
                     )*
-                    _ => unreachable!("property_info: unavailable for index {}", index),
+                    _ => unreachable!("property_info: unavailable for {}", index),
```

Comment:
Why this change? The message is less expressive now...

Reply:
oh just an accident, i initially removed it and i guess i just did something wrong when putting it back